DEVELOPER_GUIDE.md

```diff
@@ -55,7 +55,32 @@ Bound addresses will then be logged to the terminal :
 
 ## Run OpenSearch
 
-- Start a seperate terminal and navigate to the directory that OpenSearch has been cloned to using `cd OpenSearch`
+- Start a separate terminal and navigate to the directory that OpenSearch has been cloned to using `cd OpenSearch`.
+
+- Check if extensions directory exists in OpenSearch using `ls`.
+- If the directory does not exist, create it using `mkdir extensions`.
+- Navigate to the extensions folder using `cd extensions`.
+- Manually create a file titled `extensions.yml` within the extensions directory using an IDE or an in-line text editor.
+
+Sample extensions.yml file:
+
+```
+extensions:
+  - name: opensearch-sdk
+    uniqueId: opensearch-sdk-1
+    hostName: 'sdk_host'
+    hostAddress: '127.0.0.1'
+    port: '4532'
+    version: '1.0'
+    description: Extension for the Opensearch SDK Repo
+    opensearchVersion: '3.0.0'
+    javaVersion: '14'
+    className: sdk
+    customFolderName: opensearch-sdk
+    hasNativeController: false	
```

Comment:
Not for this PR: I know there are coming from `PluginInfo` but they dont make sense for extensions any more.